alternator/executor.cc

```diff
@@ -109,16 +109,20 @@ json::json_return_type make_streamed(rjson::value&& value) {
         // move objects to coroutine frame.
         auto los = std::move(os);
         auto lrs = std::move(rs);
+        std::exception_ptr ex;
         try {
             co_await rjson::print(*lrs, los);
-            co_await los.close();
         } catch (...) {
             // at this point, we cannot really do anything. HTTP headers and return code are
             // already written, and quite potentially a portion of the content data.
             // just log + rethrow. It is probably better the HTTP server closes connection
             // abruptly or something...
-            elogger.error("Unhandled exception in data streaming: {}", std::current_exception());
-            throw;
+            ex = std::current_exception();
+            elogger.error("Exception during streaming HTTP response: {}", ex);
+        }
+        co_await los.close();
```

Comment:
Good catch! I have to say I find it very surprising/upsetting that after writing to a file has already **failed**, it is still mandatory to close it otherwise you get an assertion from Seastar. It's like a bug magnet. But even if that Seastar issue was somehow solved (which it isn't), your patch is still good and important because in theory rjson::print() could have failed on something else - not just a file error.